my-app/api/user/detail.ts

```diff
@@ -0,0 +1,17 @@
+import { User } from "@/interfaces/user";
+import { ApiResponse } from "@/interfaces/api-response";
+import { apiClient } from "@/api";
+import { AxiosError, isAxiosError } from "axios";
+
+export async function detailUser(id: string): Promise<ApiResponse<User>> {
```

Comment:
```suggestion
export async function getUserDetails(id: string): Promise<ApiResponse<User>> {
```

Funções normalmente ficam melhores como verbos.